### PR TITLE
fix(iroh-net): correctly report sent transmits in poll_send

### PR DESCRIPTION
**Pull request by dignifiedquire**
If udp and derp errored (or are not available) but we have sent pings, this would wrongly report that all transmits were sent.

### CODE REVIEW

**Review by ramfox — APPROVED** [review]

File: iroh-net/src/magicsock.rs [hunk 1]
```diff
@@ -388,11 +390,10 @@ impl Inner {
                     }
                 }
 
-                let n = transmits.len();
-
                 // send derp
                 if let Some(ref derp_url) = derp_url {
                     self.try_send_derp(derp_url, public_key, split_packets(&transmits));
+                    transmits_sent = transmits.len();
```
Comment (flub) [inline iroh-net/src/magicsock.rs:396]: this seems to be mostly a wish to be fair, given how `try_send_derp` can drop things on the floor if it feels like it.  but for the purpose of this fix it's probably fine to leave as it.

File: iroh-net/src/magicsock.rs [hunk 2]
```diff
@@ -408,14 +409,18 @@ impl Inner {
...
                         send_derp = ?derp_url,
                         "sent transmits"
                     );
-                    Poll::Ready(Ok(n))
+                    assert!(
+                        transmits_sent <= transmits_len,
+                        "too many sent {} > {}",
+                        transmits_sent,
+                        transmits_len
+                    );
```
Comment (flub) [inline iroh-net/src/magicsock.rs:422]: curious why you feel the need for this assert?  It's also the only reason you keep `transmits_len` around.

**Review by flub — APPROVED** [review]

File: iroh-net/src/magicsock.rs (hunk 2 above)
Comment (dignifiedquire) [reply iroh-net/src/magicsock.rs:422]: this was from debugging this, probably can remove it

File: iroh-net/src/magicsock.rs (hunk 1 above)
Comment (flub) [reply iroh-net/src/magicsock.rs:395]: i guess that's a fair way to look at it

**Review by Frando — APPROVED** [review]

**Pull request merged**